packages/@aws-cdk/aws-events/lib/event-pattern.ts

```diff
@@ -37,15 +349,15 @@ export interface EventPattern {
    *
    * @default - No filtering on version
    */
-  readonly version?: string[];
+  readonly version?: string[] | IMatcher;
```

Comment:
Avoid using union types in L2 libraries. Even though they work fine in TypeScript, they do not behave well in many of the other languages we target (See [Design Guidelines, API Design, Props, Unions](https://github.com/aws/aws-cdk/blob/v1-main/docs/DESIGN_GUIDELINES.md#unions)).